test/unittest/base/test_document.py

```diff
@@ -65,6 +67,35 @@ def test_encode_decode_flattened_document():
     assert isinstance(next(iter(encoded_r[KEY_BLOBS].values())), bytes)
 
 
+def test_encode_model_with_remote_blob():
+    m = ObjectModel(
+        identifier='test',
+        object=lambda x: x + 2,
+    )
+
+    encoded_r = m.encode()
+    with tempfile.TemporaryDirectory() as temp_dir:
+        blob_key = [k for k in encoded_r['_builds'] if k not in ('test', 'dill_lazy')][
+            0
+        ]
+
+        blob = encoded_r['_blobs'][blob_key]
+        temp_file_path = os.path.join(temp_dir, blob_key)
+        with open(temp_file_path, 'wb') as f:
+            f.write(blob)
+
+        encoded_r['_builds'][blob_key]['blob'] = f'&:blob:file://{temp_file_path}'
+
+        def local_blob(x, loader=None):
+            return loader(x)
+
+        decoded_r = Document.decode(encoded_r, getters={'blob': local_blob})
```

Comment:
In the user interface, we should not define getters. Right? @blythed

Reply:
I copied it from below test case we are already defining the getters

Reply:
done, thank you